ashpd-demo/src/portals/desktop/usb.rs

```diff
@@ -0,0 +1,373 @@
+// Copyright (C) 2024 GNOME Foundation
+//
+// Authors:
+//     Hubert Figuière <hub@figuiere.net>
+//
+
+use std::{collections::HashMap, os::fd::AsRawFd, sync::Arc};
+
+use adw::{prelude::*, subclass::prelude::*};
+use ashpd::{
+    desktop::{
+        usb::{Device, UsbDevice, UsbError, UsbProxy},
+        Session,
+    },
+    zbus::zvariant::{Fd, OwnedFd},
+    WindowIdentifier,
+};
+use futures_util::{lock::Mutex, StreamExt};
+use glib::clone;
+use gtk::glib;
+use rusb::UsbContext;
+
+use crate::widgets::{PortalPage, PortalPageExt, PortalPageImpl};
+
+glib::wrapper! {
+    pub struct UsbDeviceRow(ObjectSubclass<imp::UsbDeviceRow>)
+        @extends gtk::Widget, gtk::ListBoxRow, adw::PreferencesRow, adw::ActionRow;
+}
+
+impl UsbDeviceRow {
+    fn with_device(page: UsbPage, device_id: String, writable: bool) -> Self {
+        let obj: Self = glib::Object::new();
+
+        let imp = obj.imp();
+        imp.page.replace(Some(page));
+        imp.device_id.replace(device_id);
+        imp.writable.set(writable);
+        obj
+    }
+
+    fn acquire(&self) {
+        self.imp().checkbox.set_active(true);
+    }
+
+    fn release(&self) {
+        self.imp().checkbox.set_active(false);
+    }
+}
+
+mod imp {
```

Comment:
This doesn't follow the current code style that puts mod imp at the top of the file